### PR TITLE
initial work for hashing passwords

### PR DESCRIPTION
**Pull request by mwj10**
Passwords are hashed when created and updated (i.e., if changed). Files affected routes, test_routes, features.

### CODE REVIEW

File: service/models.py [hunk 1]
```diff
@@ -233,16 +240,23 @@ def create(self):
...
         """
         Updates a Customer to the database
         """
         logger.info("Updating/Saving %s, %s", self.last_name, self.first_name)
         if not self.id:
             raise DataValidationError("Update called with empty ID field")
+
+        # if PWD changed, hash again
+        if original_password is not None and not original_password == self.password:
+            self.password = hash_password(self.password)
```
Comment (Akshama) [inline service/models.py:258]: Can you add a test for this line so that the coverage is maintained?

File: service/routes.py [hunk 2]
```diff
@@ -153,13 +153,16 @@ def put(self, customer_id):
         """
         app.logger.info('Request to Update a Customer with id [%s]', customer_id)
         customer = Customer.find(customer_id)
+        original_passowrd = None
```
Comment (Akshama) [inline service/routes.py:156]: ~original_passowrd~ original_password

File: tests/test_routes.py [hunk 3]
```diff
@@ -26,6 +27,9 @@
 FLAG = False
 
 
+def hash_passowrd(password):
```
Comment (Akshama) [inline tests/test_routes.py:30]: Please correct the typo. 
If there is an in-built function with the same name, you can use hash_password_util

File: service/models.py (hunk 1 above)
Comment (mwj10) [reply service/models.py:258]: Done

File: service/routes.py (hunk 2 above)
Comment (mwj10) [reply service/routes.py]: Done

File: tests/test_routes.py (hunk 3 above)
Comment (mwj10) [reply tests/test_routes.py]: Done

**Comment by Akshama** [comment]
LGTM! Merging.

**Pull request merged**